### ISSUE TITLE
Exception is created but not thrown

### ISSUE THREAD

**Issue by AppChecker** [magento/magento2#6320]
Hello!
1) In method [Collection::filterByProductType](https://github.com/magento/magento2/blob/1b30452b84a03d6951ba54ee699ee7df5ac47dd3/app/code/Magento/Reports/Model/ResourceModel/Product/Lowstock/Collection.php#L252):

```
      if (!is_string($typeFilter) && !is_array($typeFilter)) {
            new \Magento\Framework\Exception\LocalizedException(__('The product type filter specified is incorrect.'));
        }
```

\Magento\Framework\Exception\LocalizedException  is created but not thrown.
Is "throw" keyword missing?

2) In method [AbstractProduct::getModel](https://github.com/magento/magento2/blob/1b30452b84a03d6951ba54ee699ee7df5ac47dd3/app/code/Magento/Reports/Block/Product/AbstractProduct.php#L93):

```
        try {
            $model = $this->_indexFactory->get($this->_indexType);
        } catch (\InvalidArgumentException $e) {
            new \Magento\Framework\Exception\LocalizedException(__('Index type is not valid'));
        }
```

\Magento\Framework\Exception\LocalizedException  is created but not thrown.

These possible defects found by [static analyzer AppChecker](https://cnpo.ru/en/solutions/appchecker.php)

**Comment by odubovyk** [comment]
Internal ticket MAGETWO-57726

**Comment by picamator** [comment]
@AppChecker Thanks for reporting.
We've created internal ticket MAGETWO-57727 to fix it.

**Comment by magento-engcom-team** [comment]
@AppChecker, thank you for your report.
The issue is already fixed in 2.2.0

**Comment by magento-engcom-team** [comment]
@AppChecker, thank you for your report.
The issue is already fixed in 2.2.0
